src/App.jsx

```diff
@@ -1,26 +1,37 @@
+import React, { useState, useEffect } from 'react';
+import './App.css';
+import TodoList from './TodoList';
+import AddTodoForm from './AddTodoForm';
 
-import React, { useState, useEffect } from 'react'
-import './App.css'
-import TodoList from './TodoList'
-import AddTodoForm from './AddTodoForm'
-
-
-function useSemiPersistentState(key, initialState) {
-  const [value, setValue] = useState(() => {
-    const savedValue = localStorage.getItem(key);
-    return savedValue ? JSON.parse(savedValue) : initialState;
-  });
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
+  
   useEffect(() => {
-    localStorage.setItem(key, JSON.stringify(value));
-  }, [key, value]);
+    const fetchData = new Promise((resolve) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem('savedTodoList')) || [],
+          },
+        });
+      }, 2000);
+    });
+
+    fetchData.then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+    });
+  }, []);
 
-  return [value, setValue];
-}
 
+  useEffect(() => {
+    if (!isLoading) {
+      localStorage.setItem('savedTodoList', JSON.stringify(todoList));
+    }
+  }, [todoList, isLoading]);
```

Comment:
The dependency on isLoading is not needed because once the data is loaded (isLoading is false), the only thing that matters for saving to localStorage is when todoList changes. The isLoading state only controls the UI and doesn't affect the logic of saving the data.